core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,703 @@
+    await this.type(storyUrlFragmentField, storyUrlFragment);
+    await this.type(
+      storyDescriptionField,
+      `Story creation description for ${storyTitle}.`
+    );
+
+    await this.clickOn(storyPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+
+    await this.page.waitForSelector(photoUploadModal, {hidden: true});
+    await this.clickOn(createStoryButton);
+
+    await this.page.waitForSelector(storyMetaTagInput);
+    await this.page.focus(storyMetaTagInput);
+    await this.page.type(storyMetaTagInput, 'meta');
+    await this.page.keyboard.press('Tab');
+
+    await this.createChapter(explorationId);
+    await this.saveStoryDraft();
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileSaveStoryChangesDropdown);
+      await this.page.waitForSelector(mobilePublishStoryButton);
+      await this.clickOn(mobilePublishStoryButton);
+    } else {
+      await this.page.waitForSelector(`${publishStoryButton}:not([disabled])`);
+      await this.clickOn(publishStoryButton);
+      await this.page.waitForSelector(unpublishStoryButton, {visible: true});
+    }
+  }
+
+  /**
+   * Function for creating a chapter for a certain story.
+   */
+  async createChapter(explorationId: string): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileAddChapterDropdown);
+    }
+    await this.clickOn(addChapterButton);
+    await this.type(chapterTitleField, 'Test Chapter 1');
+    await this.type(chapterExplorationIdField, explorationId);
+
+    await this.clickOn(chapterPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+
+    await this.page.waitForSelector(photoUploadModal, {hidden: true});
+    await this.clickOn(createChapterButton);
+    await this.page.waitForSelector(modalDiv, {hidden: true});
+  }
+
+  /**
+   * Function for saving a story as a curriculum admin.
+   */
+  async saveStoryDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileOptionsSelector);
+      await this.clickOn(mobileSaveStoryChangesButton);
+    } else {
+      await this.clickOn(saveStoryButton);
+    }
+    await this.type(
+      saveChangesMessageInput,
+      'Test saving story as curriculum admin.'
+    );
+    await this.page.waitForSelector(`${closeSaveModalButton}:not([disabled])`);
+    await this.clickOn(closeSaveModalButton);
+    await this.page.waitForSelector(modalDiv, {hidden: true});
+  }
+
+  /**
+   * This function checks if the topic has been published successfully,
+   * by verifying the status and the counts in the topic and skills dashboard.
+   */
+  async expectPublishedTopicToBePresent(): Promise<void> {
```

Comment:
The name of this function is still not clear. When a dev reads the name of the function they should know what actions are being taken to verify, or should be able to infer that. E.g.  expectPublishedTopicToBePresentInMathClassroom() or similar.

Please clarify (and see the previous comment about going to /learn/math or the relevant topic viewer page).

Reply:
Ah, I see, sorry, you're right. OK to check via the topics and skills dashboard; in that case maybe have the name be expectTopicsAndSkillsDashboardToShowTopicIsPublished(topicName) or similar.